### PR TITLE
[Xcode12][NearbyInteraction] New framework, updated through beta 4

### PR DESCRIPTION
**Pull request by whitneyschmidt**
See https://github.com/xamarin/maccore/issues/2274 - added to Xcode 12 checklist as well.

### CODE REVIEW

File: tests/introspection/ApiCtorInitTest.cs [hunk 1]
```diff
@@ -68,6 +68,13 @@ protected virtual bool Skip (Type type)
 				return true;
 			case "NEPacketTunnelProvider":
 				return true;
+			// On iOS 14 (beta 4) we get: [NISimulator] To simulate Nearby Interaction distance and direction, launch two or more simulators and
+			// move the simulator windows around the screen.
```
Comment (rolfbjarne) [inline tests/introspection/ApiCtorInitTest.cs:72]: This sounds like a great reason to request a bigger screen :smile:

Comment (dalexsoto) [reply tests/introspection/ApiCtorInitTest.cs:72]: I got you covered https://www.samsung.com/us/computing/monitors/gaming/49--odyssey-g9-gaming-monitor-lc49g95tssnxza/ and if that is not enough you can get 2 of these https://www.staples.com/LG-86BH5C-B-86-LED-Display/product_IM12U5216

Comment (whitneyschmidt) [reply tests/introspection/ApiCtorInitTest.cs:72]: 👀 Looking for bigger apartment listings to accommodate an 86" monitor....

**Comment by mandel-macaque** [comment]
@whitneyschmidt failure is unrelated, looks like https://github.com/xamarin/maccore/issues/1028 I believe we are ok to merge the PR.

File: src/nearbyinteraction.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+namespace NearbyInteraction {
+
+    [NoWatch, NoTV, NoMac, iOS (14, 0)]
+    [Static]
+    interface NIErrorDomain
+    {
+        [Field ("NIErrorDomain")]
+        NSString Domain { get; }
+    }
```
Comment (dalexsoto) [inline src/nearbyinteraction.cs:24]: This needs to be in the `NIErrorCode` error enum like this 

https://github.com/xamarin/xamarin-macios/blob/65237b9af78aeb58095555f6a6e29b18ccb46629/src/fileproviderui.cs#L22-L29

**Pull request merged**